ui/vite.config.js

```diff
@@ -4,10 +4,44 @@ import { defineConfig } from "vite";
 import vue from "@vitejs/plugin-vue";
 import monacoEditorPlugin from "vite-plugin-monaco-editor";
 
+const streamsyncComponentsDocs = () => {
```

Comment:
The idea is amazing and the templates look so much cleaner.

This config looks a bit dirty now though. Can you please refactor by:
- Taking this to another file and importing
- Make it obvious that it's a plugin 

(similarly to `monacoEditorPlugin`)